File: perl/list_merger/list_merger_8.pl
#!/usr/bin/perl
use strict;
use warnings;

# This program merges two lists of numbers into a single sorted list.
# It demonstrates basic list operations in Perl, including concatenation,
# sorting, and printing of lists. The program is designed to help programmers
# understand how to manipulate lists and perform common operations on them.

# Define the first list of numbers
my @list1 = (3, 1, 4, 1, 5);

# Define the second list of numbers
my @list2 = (9, 2, 6, 5, 3);

# Merge the two lists into a single list
my @merged_list = (@list1, @list2);

# Sort the merged list in ascending order
@merged_list = sort { $a <=> $b } @merged_list;

# Print the sorted merged list
print "Merged and sorted list: @merged_list\n";

# Calculate the sum of the merged list
my $sum;
$sum += $_ for @merged_list;

# Print the sum of the merged list
print "Sum of merged list: $sum\n";

File: vbnet/list_merger/list_merger_8.vb
' This program is called List Merger.
' The purpose of this program is to merge two lists of integers into a single list.
' It demonstrates basic list operations such as adding elements and iterating through lists.
' The program is designed to be educational and help programmers understand list manipulation in VB.NET.

Module ListMerger

    Sub Main()
        ' Declare two lists of integers
        Dim list1 As New List(Of Integer) From {1, 2, 3, 4, 5}
        Dim list2 As New List(Of Integer) From {6, 7, 8, 9, 10}

        ' Call the MergeLists function to merge list1 and list2
        Dim mergedList As List(Of Integer) = MergeLists(list1, list2)

        ' Print the merged list
        For Each num As Integer In mergedList
            Console.WriteLine(num)
        Next

        ' Wait for user input before closing the console window
        Console.ReadLine()
    End Sub

    ' Function to merge two lists of integers
    Function MergeLists(ByVal listA As List(Of Integer), ByVal listB As List(Of Integer)) As List(Of Integer)
        ' Declare a new list to hold the merged elements
        Dim resultList As New List(Of Integer)

        ' Add elements from the first list to the result list
        For Each num As Integer In listA
            resultList.Add(num)
        Next

        ' Add elements from the second list to the result list
        For Each num As Integer In listB
            resultList.Add(num)
        Next

        ' Return the merged list
        Return resultList
    End Function

End Module

